test/test_irc.py

```diff
@@ -1,146 +1,324 @@
+
+    assert bot.backend.message_sent == rawlist('INFO')
+
+
+def test_write_args(bot):
+    bot.write(['NICK', 'Sopel'])
+
+    assert bot.backend.message_sent == rawlist('NICK Sopel')
+
+
+def test_write_text(bot):
+    bot.write(['HELP'], '?')
+
+    assert bot.backend.message_sent == rawlist('HELP :?')
+
+
+def test_write_args_text_safe(bot):
+    bot.write(['CMD\nUNSAFE'], 'Unsafe\rtext')
+
+    assert bot.backend.message_sent == rawlist('CMDUNSAFE :Unsafetext')
+
+
+def test_write_args_many(bot):
+    bot.write(['NICK', 'Sopel'])
+    bot.write(['JOIN', '#sopel'])
+
+    assert bot.backend.message_sent == rawlist(
+        'NICK Sopel',
+        'JOIN #sopel',
+    )
+
+
+def test_write_text_many(bot):
+    bot.write(['NICK', 'Sopel'])
+    bot.write(['HELP'], '?')
+
+    assert bot.backend.message_sent == rawlist(
+        'NICK Sopel',
+        'HELP :?',
+    )
+
+
+def test_action(bot):
+    bot.action('is doing some tests', '#sopel')
+
+    assert bot.backend.message_sent == rawlist(
+        'PRIVMSG #sopel :\001ACTION is doing some tests\001',
+    )
+
+
+def test_join(bot):
+    bot.join('#sopel')
+
+    assert bot.backend.message_sent == rawlist(
+        'JOIN #sopel',
+    )
+
+
+def test_join_password(bot):
+    bot.join('#sopel', 'secret_password')
+
+    assert bot.backend.message_sent == rawlist(
+        'JOIN #sopel secret_password',
+    )
+
+
+def test_kick(bot):
+    bot.kick('spambot', '#channel')
+
+    assert bot.backend.message_sent == rawlist(
+        'KICK #channel spambot',
+    )
+
+
+def test_kick_reason(bot):
+    bot.kick('spambot', '#channel', 'Flood!')
+
+    assert bot.backend.message_sent == rawlist(
+        'KICK #channel spambot :Flood!',
+    )
+
+
+def test_notice(bot):
+    bot.notice('Hello world!', '#sopel')
+
+    assert bot.backend.message_sent == rawlist(
+        'NOTICE #sopel :Hello world!',
+    )
+
+
+def test_part(bot):
+    bot.part('#channel')
+
+    assert bot.backend.message_sent == rawlist(
+        'PART #channel',
+    )
+
+
+def test_part_reason(bot):
+    bot.part('#channel', 'Bye!')
+
+    assert bot.backend.message_sent == rawlist(
+        'PART #channel :Bye!',
+    )
+
+
+def test_reply(bot):
+    bot.reply('Thank you!', '#sopel', 'dgw')
+
+    assert bot.backend.message_sent == rawlist(
+        'PRIVMSG #sopel :dgw: Thank you!',
+    )
+
+
+def test_reply_notice(bot):
+    bot.reply('Thank you!', '#sopel', 'dgw', notice=True)
+
+    assert bot.backend.message_sent == rawlist(
+        'NOTICE #sopel :dgw: Thank you!',
+    )
+
+
+def test_say(bot):
+    bot.say('Hello world!', '#sopel')
+
+    assert bot.backend.message_sent == rawlist(
+        'PRIVMSG #sopel :Hello world!',
+    )
+
+
+def test_say_safe(bot):
+    bot.say('Hello\r\nworld!\r\n', '#sopel')
+
+    assert bot.backend.message_sent == rawlist(
+        'PRIVMSG #sopel :Helloworld!',
+    )
+
+
+def test_say_long_fit(bot):
+    """Test a long message that fit into the 512 bytes limit."""
```

Comment:
```suggestion
    """Test a long message that fits into the 512 bytes limit."""
```

Aha! Found your nemesi**S** lurking!